src/i3ipc.ml

```diff
@@ -356,14 +395,33 @@ module Event = struct
     binding: binding;
   } [@@deriving of_yojson { strict = false }, show]
 
+  type shutdown_reason =
+    | Restart
+    | Exit
+  [@@deriving show]
+
+  type shutdown_event_info = {
+    change : string
+  } [@@deriving of_yojson, show]
+
+  type tick_event_info = {
+    first : bool;
+    payload : string
+  } [@@deriving of_yojson, show]
+
   type t =
     | Workspace of workspace_event_info
     | Output of output_event_info
     | Mode of mode_event_info
     | Window of window_event_info
     | BarConfig of bar_config_event_info
     | Binding of binding_event_info
+    | Shutdown of shutdown_reason
+    | Tick of tick_event_info
   [@@deriving show]
+
+  let unfold_shut_info sev =
+    sev.change
```

Comment:
I think this is superfluous (see below)

Reply:
Sometime I ask myself: why? Why I did this?